editor/reducer.js

```diff
@@ -454,6 +422,33 @@ export function preferences( state = STORE_DEFAULTS.preferences, action ) {
 				...state,
 				mode: action.mode,
 			};
+		case 'INSERT_BLOCKS':
+			// record the block usage and put the block in the recently used blocks
+			let blockUsage = state.blockUsage;
+			let recentlyUsedBlocks = [ ...state.recentlyUsedBlocks ];
+			action.blocks.forEach( ( block ) => {
+				const uses = ( blockUsage[ block.name ] || 0 ) + 1;
+				blockUsage = omit( blockUsage, block.name );
```

Comment:
Is this line necessary since we're overriding the value just after?

Reply:
I thought this is part of the principle of always creating a new values, rather than changing existing data. The state should be `deepFreeze`'d here in tests, so trying to change `blockUsage` should result in an error, because we'd be modifying the state directly.